### PR TITLE
lax.expm1: more accurate gradient for large negative inputs

### PR DESCRIPTION
**Pull request by jakevdp**
Experimental change – I want to check if this breaks anything.

Here's an example of where the current implementation of `grad(lax.expm1)` becomes inaccurate:
```python
import matplotlib.pyplot as plt
import jax.numpy as jnp
import jax

x = jnp.linspace(-30, 30, 1000)

def expm1_direct(x):
  return jnp.exp(x) - 1

fig, ax = plt.subplots(1, 2, figsize=(12, 4))
ax[0].semilogy(x, 1 + expm1_direct(x), label='1 + exp(x) - 1')
ax[0].semilogy(x, 1 + jnp.expm1(x), label='1 + expm1(x)')
ax[0].legend()

ax[1].semilogy(x, jax.vmap(jax.grad(expm1_direct))(x), label='exp(x)')
ax[1].semilogy(x, jax.vmap(jax.grad(jnp.expm1))(x), label='grad(expm1)(x)')
ax[1].legend()
```
download-1

This change makes the gradient more accurate when x is a large negative number, but it comes at the cost of re-computing `exp(x)`, so we may not want to merge (after all, the current answers do agreee to within reasonable floating point accuracy in an absolute sense, though not in a relative sense)

### CODE REVIEW

**Comment by jakevdp** [comment]
One argument against this change: the current implementation is better for in computing `grad(expm1)(x) - 1` for small values of `x`:
```python
@jax.jit
def f_direct(x):
  return jax.vmap(jax.grad(expm1_direct))(x) - 1

@jax.jit
def f_expm1(x):
  return jax.vmap(jax.grad(jnp.expm1))(x) - 1

x = jnp.logspace(-20, 20, 1000)
plt.loglog(x, f_direct(x), label='grad[exp(x) - 1] - 1')
plt.loglog(x, f_expm1(x), label='grad[expm1(x)] - 1')
plt.legend()
```
download

Merging this PR would make this expression less accurate in the small-x domain.

I'm not sure how to judge whether `grad(expm)` for $x \ll 0$ is more important than `grad(expm) - 1` for $x \to 0$
